packages/react/src/createReactQueryHooks.ts

```diff
@@ -206,59 +206,52 @@ export function createReactQueryHooks<
     return { ...hook, data };
   }
 
-  const prefetchQueryOnServer = async <
-    TPath extends keyof TQueries & string,
-    TInput extends inferProcedureInput<TQueries[TPath]>
-  >(
-    router: TRouter,
-    opts: {
-      path: TPath;
-      ctx: TContext;
-      input: TInput;
-    },
-  ): Promise<void> => {
-    const input = opts.input ?? null;
-    const { path, ctx } = opts;
-    const cacheKey = [path, input];
+  /**
+   * Create functions you can use for server-side rendering / static generation
+   * @param router Your app's router
+   * @param ctx Context used in the calls
+   */
+  function ssr(router: TRouter, ctx: TContext) {
+    const caller = router.createCaller(ctx);
+
+    const prefetchQuery = async <
+      TPath extends keyof TQueries & string,
+      TProcedure extends TQueries[TPath]
+    >(
+      ...pathAndArgs: [path: TPath, ...args: inferHandlerInput<TProcedure>]
+    ) => {
+      const [path, input] = pathAndArgs;
+      const cacheKey = [path, input ?? null];
+
+      return queryClient.prefetchQuery(cacheKey, async () => {
+        const data = await caller.query(...pathAndArgs);
 
-    await queryClient.prefetchQuery(cacheKey, async () => {
-      const data = await router.invoke({
-        target: 'queries',
-        ctx,
-        path,
-        input,
+        return data;
       });
+    };
 
-      return data;
-    });
-  };
+    const prefetchInfiniteQuery = async <
+      TPath extends keyof TQueries & string,
+      TProcedure extends TQueries[TPath]
+    >(
+      ...pathAndArgs: [path: TPath, ...args: inferHandlerInput<TProcedure>]
+    ) => {
+      const [path, input] = pathAndArgs;
+      const cacheKey = [CACHE_PREFIX_INFINITE_QUERIES, path, input ?? null];
 
-  const prefetchInfiniteQueryOnServer = async <
-    TPath extends keyof TQueries & string,
-    TInput extends inferProcedureInput<TQueries[TPath]>
-  >(
-    router: TRouter,
-    opts: {
-      path: TPath;
-      ctx: TContext;
-      input: TInput;
-    },
-  ): Promise<void> => {
-    const input = opts.input ?? null;
-    const { path, ctx } = opts;
-    const cacheKey = [CACHE_PREFIX_INFINITE_QUERIES, path, input];
+      return queryClient.prefetchInfiniteQuery(cacheKey, async () => {
+        const data = await caller.query(...pathAndArgs);
 
-    await queryClient.prefetchInfiniteQuery(cacheKey, async () => {
-      const data = await router.invoke({
-        target: 'queries',
-        ctx,
-        path,
-        input,
+        return data;
       });
+    };
 
-      return data;
-    });
-  };
+    return {
+      caller,
+      prefetchQuery,
+      prefetchInfiniteQuery,
+    };
```

Comment:
I like this. One object for all the ssr methods, then the default client thing can expose hooks